pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,610 @@
+        idx_inf, __, __ = self._check_for_inf(Zl)
+        denominator = (self.A * Zl + self.B)
+        nominator = Zl*FrequencyData(np.ones_like(denominator.freq),
+                                     self.frequencies)
+
+        # Admittance form for Zl = inf
+        nominator.freq[idx_inf] = 1
+        denominator.freq[idx_inf] = (self.A + self.B / Zl).freq[idx_inf]
+
+        # Avoid cases where denominator is zero, examples
+        # Zl = 0 & B = 0; Zl = inf & A = 0
+        denominator.freq[denominator.freq == 0] = np.finfo(float).eps
+        return nominator / denominator
+
+    def _transfer_function_q2q1(self,
+                                Zl: complex | FrequencyData) -> FrequencyData:
+        """Returns the transfer function Q2_out / Q1_in."""
+        denominator = (self.A * Zl + self.B)
+        nominator = FrequencyData(np.ones_like(denominator.freq),
+                                  self.frequencies)
+
+        # In cases where the denominator is zero, e.g. Zl = 0 & B = 0,
+        # are related to short-circuated outputs (undefined current) => NaN
+        denominator.freq[denominator.freq == 0] = np.nan
+        return nominator / denominator
+
+    def _transfer_function_q2q2(self,
+                                Zl: complex | FrequencyData) -> FrequencyData:
+        """Returns the second quantity's transfer function (Q2_out/Q2_in)."""
+        denominator = (self.C * Zl + self.D)
+        nominator = FrequencyData(np.ones_like(denominator.freq),
+                                  self.frequencies)
+
+        # Admittance form for Zl = inf
+        idx_inf, idx_inf_Zl, Zl_indexable = self._check_for_inf(Zl)
+        nominator.freq[idx_inf] = 1 / Zl_indexable[idx_inf_Zl]
+        denominator.freq[idx_inf] = (self.C + self.D / Zl).freq[idx_inf]
+
+        # Avoid cases where denominator is zero, examples
+        # Zl = 0 & D = 0; Zl = inf & C = 0
+        denominator.freq[denominator.freq == 0] = np.finfo(float).eps
+        return nominator / denominator
+
+    def _transfer_function_q1q2(self,
+                                Zl: complex | FrequencyData) -> FrequencyData:
+        """Returns the transfer function Q1_out / Q2_in."""
+        denominator = (self.C * Zl + self.D)
+        nominator = Zl*FrequencyData(np.ones_like(denominator.freq),
+                                     self.frequencies)
+
+        # Admittance form for Zl = inf
+        idx_inf, __, __  = self._check_for_inf(Zl)
+        nominator.freq[idx_inf] = 1
+        denominator.freq[idx_inf] = (self.C + self.D / Zl).freq[idx_inf]
+
+        # In cases where the denominator is zero, e.g. Zl = inf & C = 0,
+        # are related to non-physical cases (e.g. undefined current/voltage)
+        # => NaN
+        denominator.freq[denominator.freq == 0] = np.nan
+        return nominator / denominator
+
+    @classmethod
+    def _create_abcd_matrix_broadcast(cls, abcd_data: FrequencyData,
+                                      abcd_cshape : tuple):
```

Comment:
this would not be nessesaray if we apply broadcasting by definition, see my suggestions.